internal/provider/provider.go

```diff
@@ -56,18 +50,14 @@ type AlzProvider struct {
 	// provider is built and ran locally, and "test" when running acceptance
 	// testing.
 	version string
-	data    *alzProviderData
-}
-
-type AlzProviderClients struct {
-	RoleAssignmentsClient *armauthorization.RoleAssignmentsClient
+	data    *clients.Client
 }
 
-type alzProviderData struct {
-	*alzlib.AlzLib
-	mu                                   *sync.Mutex
-	clients                              *AlzProviderClients
-	suppressWarningPolicyRoleAssignments bool
+// AlzModel is the data model for the ALZ provider.
+// It embeds the generated ALZ model and the Entra authentication model.
+type AlzModel struct {
```

Comment:
I'm guessing this syntax in golang just flattens the two structs into a single struct type?